codegen/java-gen/src/main/scala/kalix/codegen/java/ComponentsSourceGenerator.scala

```diff
@@ -139,9 +140,16 @@ object ComponentsSourceGenerator {
           val messageType = component.service.messageType
           s"""@Override
              |public DeferredCall<$inputType, $outputType> $commandMethod($inputType $paramName) {
+             |  MetadataImpl met = MetadataImpl.Empty();
+             |  if (context instanceof MetadataContext) {
+             |    Metadata metadata = ((MetadataContext) context).metadata();
+             |    if (metadata.has("traceparent")) {
+             |      met.add("traceparent", metadata.get("traceparent").get());
+             |    }
```

Comment:
I don't think this conditional should go here, pull that out to the regular SDK code, and have an `additionalMetadata: Metadata` that is always included in the calls if non-empty. The less logic in templates the better.